controllers/synapse/synapse_mautrixsignal_deployment.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package synapse
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	synapsev1alpha1 "github.com/opdev/synapse-operator/apis/synapse/v1alpha1"
+)
+
+// labelsForMautrixSignal returns the labels for selecting the resources
+// belonging to the given synapse CR name.
+func labelsForMautrixSignal(name string) map[string]string {
+	return map[string]string{"app": "mautrix-signal", "synapse_cr": name}
+}
+
+// deploymentForMautrixSignal returns a Deployment object for the mautrix-signal bridge
+func (r *SynapseReconciler) deploymentForMautrixSignal(s *synapsev1alpha1.Synapse, objectMeta metav1.ObjectMeta) (client.Object, error) {
+	ls := labelsForMautrixSignal(s.Name)
+	replicas := int32(1)
+
+	// The created mautrix-signal ConfigMap Name share the same name as the
+	// mautrix-signal Deployment
+	mautrixSignalConfigMapName := objectMeta.Name
+	mautrixSignalPVCName := objectMeta.Name
+	SignaldPVCName := s.ObjectMeta.Name + "-signald"
+
+	dep := &appsv1.Deployment{
+		ObjectMeta: objectMeta,
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: ls,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: ls,
+				},
+				Spec: corev1.PodSpec{
+					// The init container is responsible of copying the
+					// config.yaml from the read-only ConfigMap to the
+					// mautrixsignal-data volume. The mautrixsignal process
+					// needs read & write access to the config.yaml file.
+					InitContainers: []corev1.Container{{
+						Image: "ubi8/ubi-minimal:8.6",
+						Name:  "initconfig",
+						VolumeMounts: []corev1.VolumeMount{{
+							Name:      "config",
+							MountPath: "/input",
+						}, {
+							Name:      "mautrixsignal-data",
+							MountPath: "/data",
```

Comment:
Until now, there was:
- a ServiceAccount for Synapse, associated to the `anyuid` SCC.
- for the Heisenbridge, we are using the `default` SA.

This MR adds a SA associated to the `anyuid` for the mautrix-signal deployment.
The signald Deployment uses the `default` SA.

The mautrixsignal image has `ENV UID=1337 GID=1337` (https://github.com/mautrix/signal/blob/master/Dockerfile#L50). Note that it is *not* using `USER 1337`.

This is the entrypoint of the mautrix-signal image: https://github.com/mautrix/signal/blob/master/docker-run.sh
Especially: `chown -R $UID:$GID /data`

When using the `default` SCC, I got some permission errors. Using `anyuid` was the simplest way to fix the issue.

Could you please advise on how to implement a better way ? Maybe just allowing this Pod to run with the `1337` user and group ? Or must this pod run as root ?

Reply:
I guess we can let it as it is for this moment. I just opened an [issue](https://github.com/opdev/synapse-operator/issues/45) to track this. In my opinion it could use the nonroot SCC with a fixed user/group and configure the Pod to have a specific user or use FSGroup as well. Even change the way the PVC is created to have its security credentials restricted. I need to verify what would be the best approach.